### PR TITLE
OS-34: Creating & Writing Stories

### PR DESCRIPTION
**Pull request by shayaanmiyy**
Navbar Component in Storybook using Bootstrap

image
Screen Shot of Storybook

image
My Local Code Structure

### CODE REVIEW

File: web/app/themes/dp-timber-theme/components/organisms/Header/Header.scss [hunk 1]
```diff
@@ -0,0 +1,5 @@
+nav {
+	img {
+		width: 140px;
```
Comment (Prashant114606) [inline web/app/themes/dp-timber-theme/components/organisms/Header/Header.scss]: can we add width attribute to img tag and see if we need this scss?

File: web/app/themes/dp-timber-theme/components/organisms/Header/Header.twig [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<nav class="navbar navbar-expand-md {{ darkTheme ? 'navbar-dark bg-dark' : 'navbar-light bg-light' }}">
+	<div class="container-fluid">
+		{% if logoImage %}
+			<img src="{{ logoImage }}" alt="Logo">
```
Comment (Prashant114606) [inline web/app/themes/dp-timber-theme/components/organisms/Header/Header.twig]: add width and height attribute to img tag

File: web/app/themes/dp-timber-theme/components/style.scss [hunk 3]
```diff
@@ -1 +1,9 @@
-// Add global styling here.
\ No newline at end of file
+// Add global styling here.
+.sb-show-main.sb-main-padded {
+	padding: 0;
+}
+
+body {
```
Comment (Prashant114606) [inline web/app/themes/dp-timber-theme/components/style.scss:2]: do we need globally or just for storybook.

File: web/app/themes/dp-timber-theme/components/style.scss [hunk 4]
```diff
@@ -1 +1,9 @@
-// Add global styling here.
\ No newline at end of file
+// Add global styling here.
+.sb-show-main.sb-main-padded {
```
Comment (Prashant114606) [inline web/app/themes/dp-timber-theme/components/style.scss]: this seems specific to the storybook can we create a separate file specific to storybook overrides?

File: web/app/themes/dp-timber-theme/components/organisms/Header/Header.twig (hunk 2 above)
Comment (Prashant114606) [reply web/app/themes/dp-timber-theme/components/organisms/Header/Header.twig]: possible to create a  image atom and use that atom here?

File: web/app/themes/dp-timber-theme/components/organisms/Header/Header.scss (hunk 1 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/components/organisms/Header/Header.scss]: Yes I can that too as well

File: web/app/themes/dp-timber-theme/components/organisms/Header/Header.twig (hunk 2 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/components/organisms/Header/Header.twig]: Yes I will create an Image Atom for it

File: web/app/themes/dp-timber-theme/components/style.scss (hunk 4 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/components/style.scss]: Okay, For Storybook css override, I will create a new File

File: web/app/themes/dp-timber-theme/components/style.scss (hunk 3 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/components/style.scss:2]: We need that globally, or how can we specify css only for storybook?

File: web/app/themes/dp-timber-theme/components/organisms/Header/Header.stories.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import header from "./Header.twig";
+// import menuData from "./menus.yml";
```
Comment (Prashant114606) [inline web/app/themes/dp-timber-theme/components/organisms/Header/Header.stories.js]: remove the dead code and also remove the unused yml file.

File: web/app/themes/dp-timber-theme/.storybook/main.js [hunk 6]
```diff
@@ -49,6 +49,10 @@ const config = {
           options: {},
         },
       ],
+    },
+    {
+      test: /\.yml$/,
```
Comment (Prashant114606) [inline web/app/themes/dp-timber-theme/.storybook/main.js]: remove this if not needed.

File: web/app/themes/dp-timber-theme/components/atoms/image/image.twig [hunk 7]
```diff
@@ -0,0 +1 @@
+<img src="{{ src }}" alt="{{ alt }}" class="{{ classes }}" width="{{ width }}" height="{{ height }}">
```
Comment (Prashant114606) [inline web/app/themes/dp-timber-theme/components/atoms/image/image.twig]: set a default class which will be a very generic class like classes|default('default_class_name')

File: web/app/themes/dp-timber-theme/components/organisms/Header/Header.stories.js (hunk 5 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/components/organisms/Header/Header.stories.js]: Yes commented code removed

File: web/app/themes/dp-timber-theme/.storybook/main.js (hunk 6 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/.storybook/main.js]: Okay Alright

File: web/app/themes/dp-timber-theme/components/atoms/image/image.twig (hunk 7 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/components/atoms/image/image.twig]: Okay Got It

File: web/app/themes/dp-timber-theme/.storybook/main.js (hunk 6 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/.storybook/main.js]: Removed

File: web/app/themes/dp-timber-theme/components/atoms/image/image.twig (hunk 7 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/components/atoms/image/image.twig]: Yes added img-fluid bootstrap class, when empty class passed img-default will be there

File: web/app/themes/dp-timber-theme/components/organisms/Header/Header.stories.js (hunk 5 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/components/organisms/Header/Header.stories.js]: Also, the yml file

File: web/app/themes/dp-timber-theme/components/organisms/Header/Header.scss (hunk 1 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/components/organisms/Header/Header.scss]: Yes I will do it

File: web/app/themes/dp-timber-theme/components/style.scss (hunk 3 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/components/style.scss:2]: yes we need that globally

**Pull request merged**